### PR TITLE
Fix binary-ops gtests coded in namespace cudf::test

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Fixes `BINARY_TEST` gtests source files coded in namespace `cudf::test`
This also partially addresses #12241 
No function or test has changed just the test source code reworked per namespaces.

Reference https://github.com/rapidsai/cudf/issues/11734

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by bdice — APPROVED** [review]
All looks good to me!

File: cpp/tests/binaryop/assert-binops.h [hunk 1]
```diff
@@ -58,6 +53,8 @@ struct stringify_out_values {
 // This comparator can be used to compare two values that are within a max ULP error.
 // This is typically used to compare floating point values computed on CPU and GPU which is
 // expected to be *near* equal, or when computing large numbers can yield ULP errors
+//
+// TODO: This should not be used in favor of the built-in one in column_utilities
```
[the remark this answers is not in the document]
Comment (davidwendt) [reply cpp/tests/binaryop/assert-binops.h:57]: This whole file may actually go away with #12241 
I'll revisit it then.

File: cpp/tests/binaryop/binop-null-test.cpp [hunk 2]
```diff
@@ -17,189 +17,122 @@
...
+  auto rhs = cudf::test::fixed_width_column_wrapper<int32_t>({0, 1, 2, 3, 4, 5, 6, 7, 8, 9});
 
-  auto out =
-    cudf::binary_operation(lhs, rhs, cudf::binary_operator::ADD, data_type(type_to_id<TypeOut>()));
+  auto out = cudf::binary_operation(
+    lhs, rhs, cudf::binary_operator::ADD, cudf::data_type(cudf::type_id::INT32));
 
-  ASSERT_BINOP<TypeOut, TypeLhs, TypeRhs>(*out, lhs, rhs, ADD());
+  auto expected = cudf::test::fixed_width_column_wrapper<int32_t>(
+    {9, 9, 9, 9, 9, 9, 9, 9, 9, 9}, cudf::test::iterators::no_nulls());
```
Comment (vuule) [inline cpp/tests/binaryop/binop-null-test.cpp:124]: I hope `binary_operator::ADD` is not implemented as `{return 9;}` :P

Side question - how is `fixed_width_column_wrapper` created with `no_nulls()` different from one created without the null mask iterator parameter?

**Review by vuule — APPROVED** [review]

File: cpp/tests/binaryop/binop-null-test.cpp (hunk 2 above)
Comment (davidwendt) [reply cpp/tests/binaryop/binop-null-test.cpp:124]: The `no_nulls()` creates the null-mask with all valid bits.  Without the iterator, you get a null null-mask -- no null-mask at all.

**Pull request merged**